h2/src/main/org/h2/command/Parser.java

```diff
@@ -4351,12 +4352,30 @@ private Column parseColumnWithType(String columnName) {
                 original += " VARYING";
             }
         } else if (readIf("TIME")) {
+            if (readIf("(")) {
+                originalScale = readPositiveInt();
+                if (originalScale > ValueTime.MAXIMUM_SCALE) {
+                    throw DbException.get(ErrorCode.INVALID_VALUE_SCALE_PRECISION, Integer.toString(originalScale));
+                }
+                read(")");
+            }
             if (readIf("WITHOUT")) {
                 read("TIME");
                 read("ZONE");
                 original += " WITHOUT TIME ZONE";
             }
         } else if (readIf("TIMESTAMP")) {
+            if (readIf("(")) {
+                originalScale = readPositiveInt();
+                // Allow non-standard TIMESTAMP(..., ...) syntax
```

Comment:
which DB is this for?

Reply:
I don't know. H2 had such support before, see removed lines below:
https://github.com/h2database/h2database/blob/aa9191c55ba6a0f44551ec4ab3c58973a2f06c2b/h2/src/main/org/h2/command/Parser.java#L4439-L4440
and this test
https://github.com/h2database/h2database/blob/aa9191c55ba6a0f44551ec4ab3c58973a2f06c2b/h2/src/test/org/h2/test/scripts/testScript.sql#L6955-L6956